### PR TITLE
WIP: Add SDK migration page

### PR DESCRIPTION
**Pull request by reddvl1980**

### CODE REVIEW

File: docs/voice/voice-ios-android-cloud-migration.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+---
+title: Sinch Voice and Video SDK Migration Guide
+excerpt: >-
+  Follow this migration guide to start using modern cloud-based Sinch mobile Voice and Video SDK.
+hidden: true
+---
+
+## Intended Audience
+
+This guide serves the purpose of facilitating the ease of migration from the old, legacy Sinch Moble SDKs to the new cloud-based platform. We assume that you're familiar with key concepts, operations and glossary of the Sinch Voice and Video SDK. Here we provide a comprehensive list of the _changes_ required both in the application _and_ on your backend to get started with the new cloud-based Sinch mobile SDKs.
```
Comment (christoffer-a-sinch) [inline docs/voice/voice-ios-android-cloud-migration.md:10]: "purpose of facilitating the ease" is excessive wording. Keep it simple. E.g. "This migration guide provides an overview of adaptions required to migrate from the Sinch legacy platform to the Sinch cloud-native platform)"

Typo "Moble"

We should generally not refer to it as "cloud platform" (as soon as we deprecate the legacy one, there will be only one and it should not / will not be referred to as "cloud"). This means we should also when referring to the new SDKs, we should reference them by major version, e.g. Sinch Android 4.x SDK and Sinch iOS 5.x SDK.

File: docs/voice/voice-ios-android-cloud-migration.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+Otherwise, if you had never used Sinch Voice and Video SDK before, please refer to the respective platform documentation:
+- [Sinch cloud-based Android SDK v.4](doc:voice-for-android-cloud)
```
Comment (christoffer-a-sinch) [inline docs/voice/voice-ios-android-cloud-migration.md:13]: Remove "cloud-based". 

Be consistent how we reference versions. not `v.`. Use `4.x` (no v)

File: docs/voice/voice-ios-android-cloud-migration.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+## General Changes
+
+The changes can be grouped by the platform (iOS/Android) or along the client/server boundaries. The table below provides a good eagle-eye overview of them:
+
+|                                                               | Client    | Backend   |  Android | iOS|
+| --------------------------------------------------------------|-----------|-----------|----------| ---|
+|User Registration Changes|v|v|v|v|
+|Backend User Authentication Server||v|v|v|
+|UserController API|v||v||
+|Huawei Push Notifications|v||v||
```
Comment (christoffer-a-sinch) [inline docs/voice/voice-ios-android-cloud-migration.md:34]: Support for Huawei Push is not a change, it is a new feature. Should not be in this migration doc at all.

File: docs/voice/voice-ios-android-cloud-migration.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+- [Build and Dependencies](doc:voice-ios-android-cloud-migration#build-and-dependencies)
+
+## General Changes
+
+The changes can be grouped by the platform (iOS/Android) or along the client/server boundaries. The table below provides a good eagle-eye overview of them:
+
+|                                                               | Client    | Backend   |  Android | iOS|
+| --------------------------------------------------------------|-----------|-----------|----------| ---|
+|User Registration Changes|v|v|v|v|
+|Backend User Authentication Server||v|v|v|
```
Comment (christoffer-a-sinch) [inline docs/voice/voice-ios-android-cloud-migration.md:32]: Don't introduce new concepts. We don't use the term  "Backend User Authentication Server", or "User Authentication Server", or similar. The primary terminology we should use is "User Authentication & Authorization" and "Token-based User Registration"

File: docs/voice/voice-ios-android-cloud-migration.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+|UserController API|v||v||
+|Huawei Push Notifications|v||v||
+|Backend OAuth Server for Huawei Push ||v|v||
+
+## Common Registration Changes
```
Comment (christoffer-a-sinch) [inline docs/voice/voice-ios-android-cloud-migration.md:37]: Too vague title. Common in what sense. Registration of what. Suggestion: "Client / User Registration Changes"

File: docs/voice/voice-ios-android-cloud-migration.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+When you initiate `SinchClient` you have to provide _user identity_. In the legacy Sinch SDKs, user authentication and the authorization of the registration was based on credentials provided in a way of an opaque token, based on the _SHA1_ digest of Sinch _Application Secret_. There was also a way to provide _Application Secret_ directly to the _SinchClient_.
+
+In the new SDK, registration is made both more secure and transparent:
```
Comment (christoffer-a-sinch) [inline docs/voice/voice-ios-android-cloud-migration.md:41]: "transparent" adds no information here. If you have JWT usage in mind, consider using "simplified".

File: docs/voice/voice-ios-android-cloud-migration.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+- _Application Secret_ must never be stored in the application, and there is no way to provide _Application Secret_ to the _SinchClient_ that could encourage that, instead:
+- Authorization is granted by providing industry-standard JWT registration token.
```
Comment (christoffer-a-sinch) [inline docs/voice/voice-ios-android-cloud-migration.md:43]: Wording here implies that there is something like an "industry standard registration token". JWT can be argued one standard yes, but not a "registration token". 

Suggestion: "Authentication and Authorization is granted by JWT tokens."

File: docs/voice/voice-ios-android-cloud-migration.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+> Though the use of _UserController_ is __optional__ for the FCM Push Notification case, since the _SinchClient_ can 
+> register itself to receive FCM Push, it is __mandatory__ for the Huawei Push Notifications, since the _push device
+> token_ have to be acquired by the _Application_ before constructing _UserController_.
+> Thus, it is __highly recommended__ to use _UserController_ in both cases.
+
+The action flow diagram of the _User_ registration via _UserController_ is provided below. _UserController_'s callbacks are highlighted in pale blue.
+
+![Registering User via UserController](voice-for-android-cloud\images\usercontroller-callbacks.pu.png)
+
+## Backend OAuth Server for Huawei Push
```
Comment (christoffer-a-sinch) [inline docs/voice/voice-ios-android-cloud-migration.md:71]: Remove

File: docs/voice/voice-ios-android-cloud-migration.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+> ❗️ 
+> MISSING SECTION (Christoffer)
+
```
Comment (christoffer-a-sinch) [inline docs/voice/voice-ios-android-cloud-migration.md:75]: We should add section on iOS APNs Signing Keys

File: docs/voice/voice-ios-android-cloud-migration.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+Both [iOS](doc:voice-ios-cloud-auth) and [Android](doc:voice-android-cloud-application-authentication) documents provide the same coverage on the formation of the JWT registration token, including links to the server-side sample code.
+
+## Backend User Authentication Server
+
+> ❗️ 
+> MISSING SECTION (Christoffer)
+
+## Android and UserController
+
+There is a new way to register _User_ on Android: _UserController API_. [UserController](voice-for-android-cloud/reference/com/sinch/android/rtc/UserController.html) provides a way independently from the `SinchClient` lifecycle, to register a _User_ for incoming calls via push notifications. You can also use it to un-register push token if receiving of incoming calls is no longer desirable (e.g. on logout, or changing users).
```
Comment (christoffer-a-sinch) [inline docs/voice/voice-ios-android-cloud-migration.md:54]: "unregister", not "un-register"

File: docs/voice/voice-ios-android-cloud-migration.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+> ❗️
+>
+> Though the use of _UserController_ is __optional__ for the FCM Push Notification case, since the _SinchClient_ can 
+> register itself to receive FCM Push, it is __mandatory__ for the Huawei Push Notifications, since the _push device
+> token_ have to be acquired by the _Application_ before constructing _UserController_.
+> Thus, it is __highly recommended__ to use _UserController_ in both cases.
+
+The action flow diagram of the _User_ registration via _UserController_ is provided below. _UserController_'s callbacks are highlighted in pale blue.
+
+![Registering User via UserController](voice-for-android-cloud\images\usercontroller-callbacks.pu.png)
```
Comment (christoffer-a-sinch) [inline docs/voice/voice-ios-android-cloud-migration.md:69]: This new diagram should be added to the the regular User Controller sections as well?

Comment (reddvl1980) [reply docs/voice/voice-ios-android-cloud-migration.md:56]: added

File: docs/voice/voice-ios-android-cloud-migration.md (hunk 10 above)
Comment (reddvl1980) [reply docs/voice/voice-ios-android-cloud-migration.md]: fixed

**Pull request merged**